_posts/2020-11-09-the-cloud-data-warehouse.md

```diff
@@ -0,0 +1,158 @@
+
+This blog post is in two parts; in the first part, we will elaborate on modern Cloud Data Warehouses; in the second part we will focus on how their features allow a more flexible and agile way of delivering analytics engineering, as described in the Analytics Guide Book.
+
+## The Cloud Data Warehouse
+
+A Data Warehouse is an **enabling technology** for your business to **make decisions** in relation to **the data you gather**. By arranging, conforming, and curating your data to radiate the key performance indicators of your business, a Data Warehouse forms one half of *the presentation layer* - alongside visualisation tools - that enable users in your organisation to identify insights. At least, this is the **traditional view**.
+
+Modern Cloud Data Warehouses can support the existing status quo, including various data modelling methodologies such as Kimball, Inmon or Data Vault for example. Additionally, sticking the word ***cloud*** in front of *data warehouse* does what you would expect cloud to do, in addressing many of the technological limitations or complications associated with traditional Data Warehouses - namely storage availability, compute scalability, and high availability.
+
+However, modern Cloud Data Warehouse offerings typically go far beyond that. Whilst delivering the core competencies of what a Data Warehouse should offer, they often extend these capabilities with features not readily seen before. For example:
+
+* [BigQuery ML](https://cloud.google.com/bigquery-ml/docs/introduction) allows users to inference with Machine Learning models against their datasets directly, using SQL. A more generic feature is also available on Snowflake using [External Functions](https://docs.snowflake.com/en/sql-reference/external-functions-introduction.html) 
+* Snowflake's [Zero Copy Data Sharing](https://docs.snowflake.com/en/user-guide-data-share.html) across accounts in the same cloud region, and [Replication Across Cloud Platforms](https://docs.snowflake.com/en/user-guide/database-replication-failover.html) as described in our recent blog post on [Snowflake Organizations](https://mechanicalrock.github.io/2020/09/23/snowflake-org-structures.html), are a powerful enabler for complex organisations, especially those navigating multi cloud strategies
+* Data Time Travel features of both [Snowflake](https://docs.snowflake.com/en/user-guide/data-time-travel.html) and [BigQuery](https://cloud.google.com/bigquery/docs/time-travel), allowing you to see data as it was at a particular point in time
+
+As many organisations evolve to consist of more multi-disciplinary teams, so too has technology adapted to more simply support the kind of collaboration and critical insight work that most organisations require. And so, a *modernised view* of the Data Warehouse is upon us where the technology is no longer simply a keystone of the architecture, but rather a platform for enhancement and enablement of data.
+
+### Resilient Architecture
```

Comment:
Feel like you buried the point of this paragraph? 

"there is a good chance it will be different and easier to automate" leaves me absolutely cold. DR will be different seems to the point. 

Can you spell out different? Can you make the second bullet point list a list of the concerns you do have to deal with? 

I think the point you want to make is in the old world you get lost in the weeds of pointless tech detail, but in the new world you only have to consider how to leverage the new architectures to meet your business availability goals - or budget. 

At the moment this section says not much.

Reply:
i think i was trying to say, instead of taking your existing DR/BCP to the cloud, re-evaluate your risk profile and understand what kind of scenario you need to cater for.

the implication i also had in mind was that if you need the extreme of supporting high availability across regions, then you need to be prepared to be able to rebuild your data warehouse from scratch based on data at rest (e.g. s3)

re the comments about bullet points, i was wary about bullet point fatigue. there are already a lot in this article